spec/gateways/logging_api_spec.rb

```diff
@@ -0,0 +1,65 @@
+describe Gateways::LoggingApi do
+  describe '#search' do
+    subject { described_class.new }
+
+    context "with correct data" do
+      let(:username) { "AAAAA" }
+      let(:logs) do
+        [
+          {
+            "ap" => "",
+            "building_identifier" => "",
+            "id" => 1,
+            "mac" => "",
+            "siteIP" => "",
+            "start" => "2018-10-01 18:18:09 +0000",
+            "stop" => nil,
+            "success" => true,
+            "username" => username
+          },
+          {
+            "ap" => "",
+            "building_identifier" => "",
+            "id" => 1,
+            "mac" => "",
+            "siteIP" => "",
+            "start" => "2018-10-01 18:18:09 +0000",
+            "stop" => nil,
+            "success" => true,
+            "username" => username
+          }
+        ]
+      end
+
+      before do
+        endpoint = SITE_CONFIG['logging_api_search_endpoint'] + username
+        stub_request(:get, endpoint)
+          .with(headers: {
+              'Accept' => '*/*',
+              'Accept-Encoding' => 'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+              'Host' => 'api-elb.london.wifi.service.gov.uk:8443',
+              'User-Agent' => 'Ruby'
+            })
+          .to_return(status: 200, body: logs.to_json, headers: {})
```

Comment:
More of a general testing comment, but feels like the feature spec covers this pretty well.  Not sure if there is a general consensus around testing feature vs gateway, without too much overlap.  Keen to hear your thoughts

Reply:
I'm rather new to Clean Architecture so I'm happy to follow advice. If its not needed, then I'm happy to remove it. Currently we test all gateways.